File: 백준/Silver/11441. 합 구하기/합 구하기.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

     
        int n = Integer.parseInt(br.readLine());
        int[] nums = new int[n];
        StringTokenizer st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; i++) {
            nums[i] = Integer.parseInt(st.nextToken());
        }

        int m = Integer.parseInt(br.readLine());

        int[] prefix = new int[n + 1];
        for (int i = 1; i <= n; i++) {
            prefix[i] = prefix[i - 1] + nums[i - 1];
        }

        for (int query = 0; query < m; query++) {
            st = new StringTokenizer(br.readLine());
            int i = Integer.parseInt(st.nextToken());
            int j = Integer.parseInt(st.nextToken());

            int hab = prefix[j] - prefix[i - 1];
            sb.append(hab).append("\n");
        }

        System.out.print(sb);
    }
}

File: 백준/Silver/11441. 합 구하기/합 구하기.py
import sys
from bisect import bisect_left
input = sys.stdin.readline

n = int(input())
nums = list(map(int,input().split()))
m = int(input())
prefix =[0] *(n+1)
for i in range(1,n+1):
    prefix[i] = prefix[i-1]+nums[i-1]
for _ in range(m):
    i,j = map(int,input().split())
    hab = prefix[j] -prefix[i-1]
    print(hab)
